hist/histv7/inc/ROOT/RRegularAxis.hxx

```diff
@@ -0,0 +1,93 @@
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+#ifndef ROOT_RRegularAxis
+#define ROOT_RRegularAxis
+
+#include "RLinearizedIndex.hxx"
+
+#include <cstddef>
+
+namespace ROOT {
+namespace Experimental {
+
+/**
+A regular axis with equidistant bins in the interval \f$[fLow, fHigh)\f$.
+
+For example, the following creates a regular axis with 10 normal bins between 5 and 15:
+~~~ {.cxx}
+ROOT::Experimental::RRegularAxis axis(10, 5, 15);
+~~~
+
+It is possible to disable underflow and overflow bins by passing `enableFlowBins = false`. In that case, arguments
+outside the axis will be silently discarded.
+
+\warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback is
+welcome!
+*/
+class RRegularAxis final {
+   /// The number of normal bins
+   std::size_t fNumNormalBins;
+   /// The lower end of the axis interval
+   double fLow;
+   /// The upper end of the axis interval
+   double fHigh;
+   /// The cached inverse of the bin width to speed up ComputeLinearizedIndex
+   double fInvBinWidth; //!
+   /// Whether underflow and overflow bins are enabled
+   bool fEnableFlowBins;
+
+public:
+   /// Construct a regular axis object.
+   ///
+   /// \param[in] numNormalBins the number of normal bins
+   /// \param[in] low the lower end of the axis interval (inclusive)
+   /// \param[in] high the upper end of the axis interval (exclusive)
+   /// \param[in] enableFlowBins whether to enable underflow and overflow bins
+   RRegularAxis(std::size_t numNormalBins, double low, double high, bool enableFlowBins = true)
```

Comment:
One could possibly make this `constexpr`, in case some users want to predefine an axis and reuse it several times.

Don't know if this helps in any way, though.

Reply:
Yes, I think we could for `RRegularAxis` but it's not clear to me how much this would help; users can already cheaply copy the class. It doesn't work for `RVariableBinAxis` until C++20 makes `std::vector` constructors `constexpr`. So I think for symmetry and lack of strong motivation, I would keep it without the annotations for the moment.